### PR TITLE
[ICD] Add units to configs and variables 

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
Add units to the name of the different configurations and variables.
Addresses https://github.com/project-chip/connectedhomeip/pull/27272#discussion_r1334399174

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/icd/IcdManagementServer.h [hunk 1]
```diff
@@ -30,21 +30,11 @@ using chip::Protocols::InteractionModel::Status;
...
-        mIdleInterval           = idle_interval;
-        mActiveInterval         = active_interval;
-        mActiveThreshold        = active_threshold;
-        mIcdCounter             = count;
-        mFabricClientsSupported = fabric_clients_supported;
-    }
+    uint32_t GetIdleModeInterval() { return mIdleInterval_s; }
 
-    uint32_t GetIdleModeInterval() { return mIdleInterval; }
+    uint32_t GetActiveModeInterval() { return mActiveInterval_ms; }
```
Comment (andy31415) [inline src/app/icd/IcdManagementServer.h:35]: Technically methods should have unit as well (`GetActiveModeIntervalMs`).

File: src/app/icd/IcdManagementServer.h [hunk 2]
```diff
@@ -67,20 +57,21 @@ class IcdManagementServer
...
+    uint32_t mIdleInterval_s = CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL_SEC; // in seconds.
 
-    static_assert((CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL) <= (CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL * 1000),
+    static_assert((CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL_MS) <= (CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL_SEC * 1000),
                   "Spec requires the IdleModeInterval be equal or greater to the ActiveModeInterval.");
-    static_assert((CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL) >= 300,
+    static_assert((CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL_MS) >= 300,
                   "Spec requires the ActiveModeThreshold to be equal or greater to 300ms");
-    uint32_t mActiveInterval = CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL; // in milliseconds
+    uint32_t mActiveInterval_ms = CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL_MS; // in milliseconds
```
Comment (andy31415) [inline src/app/icd/IcdManagementServer.h:70]: comments of in milliseconds/in seconds do not seem needed anymore

**Review by andy31415 — APPROVED** [review]
Thank you!

File: src/app/icd/IcdManagementServer.h (hunk 1 above)
Comment (mkardous-silabs) [reply src/app/icd/IcdManagementServer.h]: done

File: src/app/icd/IcdManagementServer.h (hunk 2 above)
Comment (mkardous-silabs) [reply src/app/icd/IcdManagementServer.h]: done

File: src/app/icd/IcdManagementServer.h [hunk 3]
```diff
@@ -67,20 +57,21 @@ class IcdManagementServer
...
+    static_assert((CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL_SEC) <= 64800,
                   "Spec requires the IdleModeInterval to be equal or inferior to 64800s.");
-    static_assert((CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL) >= 1, "Spec requires the IdleModeInterval to be equal or greater to 1s.");
-    uint32_t mIdleInterval = CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL; // in seconds.
+    static_assert((CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL_SEC) >= 1,
+                  "Spec requires the IdleModeInterval to be equal or greater to 1s.");
+    uint32_t mIdleInterval_s = CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL_SEC;
 
-    static_assert((CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL) <= (CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL * 1000),
+    static_assert((CHIP_CONFIG_ICD_ACTIVE_MODE_INTERVAL_MS) <= (CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL_SEC * 1000),
```
Comment (jtung-apple) [inline src/app/icd/IcdManagementServer.h:66]: Consider using `kMillisecondsPerSecond` from TimeUtils.h

File: src/app/tests/TestICDManager.cpp [hunk 4]
```diff
@@ -114,20 +114,20 @@ class TestICDManager
 
         // After the init we should be in active mode
         NL_TEST_ASSERT(aSuite, ctx->mICDManager.mOperationalState == ICDManager::OperationalState::ActiveMode);
-        AdvanceClockAndRunEventLoop(ctx, IcdManagementServer::GetInstance().GetActiveModeInterval() + 1);
+        AdvanceClockAndRunEventLoop(ctx, IcdManagementServer::GetInstance().GetActiveModeIntervalMs() + 1);
         // Active mode interval expired, ICDManager transitioned to the IdleMode.
         NL_TEST_ASSERT(aSuite, ctx->mICDManager.mOperationalState == ICDManager::OperationalState::IdleMode);
-        AdvanceClockAndRunEventLoop(ctx, (IcdManagementServer::GetInstance().GetIdleModeInterval() * 1000) + 1);
+        AdvanceClockAndRunEventLoop(ctx, (IcdManagementServer::GetInstance().GetIdleModeIntervalMs() * 1000) + 1);
```
Comment (jtung-apple) [inline src/app/tests/TestICDManager.cpp:120]: Consider using `kMillisecondsPerSecond` from TimeUtils.h

**Review by jtung-apple — APPROVED** [review]

File: src/app/tests/TestICDManager.cpp (hunk 4 above)
Comment (jtung-apple) [inline src/app/tests/TestICDManager.cpp:120]:
```suggestion
        AdvanceClockAndRunEventLoop(ctx, (IcdManagementServer::GetInstance().GetIdleModeIntervalSec() * 1000) + 1);
```

Shouldn't this be `GetIdleModeIntervalSec()`?

Comment (mkardous-silabs) [reply src/app/tests/TestICDManager.cpp:120]: for some reason, my search and replace didn't work as expected.

Comment (mkardous-silabs) [reply src/app/tests/TestICDManager.cpp]: use the inline function instead

File: src/app/icd/IcdManagementServer.h (hunk 3 above)
Comment (mkardous-silabs) [reply src/app/icd/IcdManagementServer.h]: done

**Pull request merged**